platform/android/java_godot_lib_jni.cpp

```diff
@@ -1352,6 +1367,52 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_method(JNIEnv *env, j
 	s->add_method(mname, mid, types, get_jni_type(retval));
 }
 
+JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_addsignal(JNIEnv *env, jobject obj, jstring sname, jstring name, jobjectArray args) {
+
+	String singname = jstring_to_string(sname, env);
+
+	ERR_FAIL_COND(!jni_singletons.has(singname));
+
+	JNISingleton *s = jni_singletons.get(singname);
+
+	String mname = jstring_to_string(name, env);
+	Vector<Variant::Type> types;
+
+	int stringCount = env->GetArrayLength(args);
+
+	for (int i = 0; i < stringCount; i++) {
+
+		jstring string = (jstring)env->GetObjectArrayElement(args, i);
+		const String rawString = jstring_to_string(string, env);
+		types.push_back(get_jni_type(rawString));
+	}
+
+	s->add_signal(mname, types);
+}
+
+JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_emitsignal(JNIEnv *env, jobject obj, jstring sname, jstring name, jobjectArray params) {
+	String singname = jstring_to_string(sname, env);
+
+	ERR_FAIL_COND(!jni_singletons.has(singname));
+
+	JNISingleton *s = jni_singletons.get(singname);
+
+	String mname = jstring_to_string(name, env);
+
+	int count = env->GetArrayLength(params);
+	Variant args[VARIANT_ARG_MAX];
+
+	for (int i = 0; i < MIN(count, VARIANT_ARG_MAX); i++) {
+
+		jobject obj = env->GetObjectArrayElement(params, i);
+		if (obj)
+			args[i] = _jobject_to_variant(env, obj);
+		env->DeleteLocalRef(obj);
+	};
+
+	s->emit_signal(mname, args[0], args[1], args[2], args[3], args[4]);
```

Comment:
Wouldn't the second version of `emit_signal` be more appropriate here?
```
Error emit_signal(const StringName &p_name, const Variant **p_args, int p_argcount);
```

Reply:
Sure!

Reply:
Now it shows me warning:
```
platform/android/java_godot_lib_jni.cpp:1406:24: warning: address of array 'args' will always evaluate to 'true' [-Wpointer-bool-conversion]
        s->emit_signal(mname, args, MIN(count, VARIANT_ARG_MAX));
```
As far as I can see it doest work. My signal from firebase_auth didn't receive because:
```
                        E  **ERROR**: Error calling method from signal 'logged_in': 'Node(auth.gd)::_on_logged_in': Method expected 0 arguments, but called with 2..
                         E     At: core/object.cpp:1228:emit_signal() - Error calling method from signal 'logged_in': 'Node(auth.gd)::_on_logged_in': Method expected 0 argum
                            ents, but called with 2..
```
But I call it with 0 (zero) arguments.